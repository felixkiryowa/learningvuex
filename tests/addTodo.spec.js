import {
    shallowMount,
    createLocalVue
} from '@vue/test-utils';
import Vuex from 'vuex';
import AddTodo from '../src/components/AddTodo.vue';
import {
    __createMocks as createStoreMocks
} from '../src/store';


// Tell Jest to use the mock
// Implementation of the store

jest.mock('../src/store/index.js');

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Testing TODOS component', () => {
    let storeMocks;
    let wrapper;


    beforeEach(() => {

        storeMocks = createStoreMocks();
        wrapper = shallowMount(AddTodo, {
            store: storeMocks.store,
            localVue,
        });
    });

    it("returns an error for a valid code", () => {
        expect(wrapper).toMatchSnapshot();
      });


    it("sets the value", () => {
        const input = wrapper.find('#title')

        input.setValue('Post One')

        expect(input.element.value).toBe('Post One')
    });

    it('It should add  to do when clicked.', () => {
        wrapper.find('#submitBtn').trigger('submit');

        expect(storeMocks.actions.addTodo).toHaveBeenCalled();
    });

});