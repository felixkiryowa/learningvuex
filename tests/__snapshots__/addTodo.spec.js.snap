// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TODOS component returns an error for a valid code 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
